test-wc.html

```diff
@@ -1,27 +1,60 @@
 <html>
-<head>
-    <meta charset="UTF-8">
-    <meta name="viewport"
-          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
-    <meta http-equiv="X-UA-Compatible" content="ie=edge">
-    <link rel="stylesheet" href="./dist/web-components/style.css">
+  <head>
+    <meta charset="UTF-8" />
+    <meta
+      name="viewport"
+      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
+    />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <link rel="stylesheet" href="./dist/web-components/style.css" />
     <title>Document</title>
-</head>
-<body>
-
+  </head>
+  <body>
     <script src="./dist/web-components/index.umd.js"></script>
 
-    <thumbnail-panel id="tp" iiif-content="https://digirati-co-uk.github.io/wunder.json"></thumbnail-panel>
+    <button id="thumbnail-panel-next">Next</button>
+    <button id="thumbnail-panel-previous">Previous</button>
 
+    <thumbnail-panel id="tp" iiif-content="https://digirati-co-uk.github.io/wunder.json"></thumbnail-panel>
 
     <script>
-        const tp = document.getElementById('tp');
+      const tp = document.getElementById('tp');
+      const nextButton = document.getElementById('thumbnail-panel-next');
+      const previousButton = document.getElementById('thumbnail-panel-previous');
+
+      tp.addEventListener('resource-changed', (e) => {
+        const target = e.target.querySelector('[thumbnail-panel]');
+
+        if (target) {
+          const next = target.getAttribute('data-next-resource');
```

Comment:
This is a cool way of doing it! We can add functions/helpers to the web component itself in the register public api here: https://github.com/IIIF-Commons/thumbnail-panel/blob/main/src/web-components/thumbnail-panel.tsx#L23

So we could add a `getNextCanvasId()` and `getPreviousCanvasId()` that essentially does the same thing.

The `component` is the `e.target` so the code would be very similar.

Reply:
We could also add a [useImperativeHandle](https://react.dev/reference/react/useImperativeHandle) to the React component, and then hook them up in the web component.

That would allow you to do:
```tsx
const ref = useRef();

ref.getNextCanvasId(); // or getNextResourceId() or whatever you like

return <ThumbnailPanel ref={ref} ... />
```
And have it controlled in that way from the outside.

Reply:
Love it! Yes, we're going to keep exploring this tomorrow. I'll dive deeper into leveraging useRef tomorrow.